### PR TITLE
Added printTreeRecursively overload with ostream parameter

### LINKED ISSUE #261: printTreeRecursively with custom ostream
Is there a way to call `BT::printTreeRecursively(tree.rootNode())` with custom `std::ostream`? It would be useful to use a custom `std::ostream` when a process does not use `std::cout` for logging such as for a ROS node.

### PR DESCRIPTION
**Pull request by asasine**
Fixes #261

### CODE REVIEW

File: include/behaviortree_cpp_v3/behavior_tree.h [hunk 1]
```diff
@@ -56,10 +57,15 @@ void applyRecursiveVisitor(const TreeNode* root_node,
...
 /**
- * Debug function to print on screen the hierarchy of the tree.
+ * Debug function to print on screen the hierarchy of the tree to std::cout.
  */
 void printTreeRecursively(const TreeNode* root_node);
 
+/**
+ * Debug function to print the hierarchy of the tree to a custom stream.
+ */
+void printTreeRecursively(const TreeNode* root_node, std::ostream& stream);
```
Comment (facontidavide) [inline include/behaviortree_cpp_v3/behavior_tree.h:67]: can't we just have 

> void printTreeRecursively(const TreeNode* root_node, std::ostream& stream = std::cout );

For simplicity?

**Review by facontidavide — COMMENTED** [review]
Nice. Thanks for contributing. I added a comment

File: include/behaviortree_cpp_v3/behavior_tree.h (hunk 1 above)
Comment (asasine) [reply include/behaviortree_cpp_v3/behavior_tree.h]: Good idea, changed it to a default argument

**Comment by facontidavide** [comment]
Awesome, thanks again

**Pull request merged**